driver-core/src/test/unit/com/mongodb/internal/connection/PowerOfTwoBufferPoolTest.java

```diff
@@ -73,4 +74,18 @@ public void testHugeBufferRequest() {
         buf.release();
         assertNotSame(buf, pool.getBuffer((int) Math.pow(2, 10) + 1));
     }
+
+    // Racy test
+    @Test
+    public void testPruning() throws InterruptedException {
+        try (PowerOfTwoBufferPool pool = new PowerOfTwoBufferPool(10, 5, TimeUnit.MILLISECONDS)) {
+            pool.enablePruning();
+            ByteBuf byteBuf = pool.getBuffer(256);
+            ByteBuffer wrappedByteBuf = byteBuf.asNIO();
+            byteBuf.release();
+            Thread.sleep(50);
+            ByteBuf newByteBuf = pool.getBuffer(256);
+            assertNotSame(wrappedByteBuf, newByteBuf.asNIO());
```

Comment:
It may be worth also checking `assertSame`:
```java
ByteBuf byteBuf = pool.getBuffer(256);
ByteBuffer wrappedByteBuf = byteBuf.asNIO();
byteBuf.release();

byteBuf = pool.getBuffer(256);
assertSame(wrappedByteBuf, byteBuf.asNIO());
byteBuf.release();

Thread.sleep(50);
byteBuf = pool.getBuffer(256);
assertNotSame(wrappedByteBuf, byteBuf.asNIO());
```

Reply:
I didn't do this because that behavior is already tested in `testReuse`

Reply:
Right, sorry for not checking that we have a test for this behavior.